### PR TITLE
Add pageflow example

### PR DESCRIPTION
**Pull request by meiliang86**

### CODE REVIEW

File: cmd/samples/pageflow/main.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	}
+
+	return &result
+}
+
+func assertState(expected, actual State) {
+	if expected != actual {
+		message := fmt.Sprintf("Workflow in wrong state. Expected %v Actual %v", expected, actual)
+		panic(message)
+	}
```
Comment (yux0) [inline cmd/samples/pageflow/main.go:49]: Do we want to print something to indicate the workflow is making progress with expected values?

**Review by yux0 — APPROVED** [review]

File: cmd/samples/pageflow/main.go (hunk 1 above)
Comment (meiliang86) [reply cmd/samples/pageflow/main.go:49]: Sure we can do that.

File: cmd/samples/pageflow/main.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+		Logger:       h.Logger,
+	}
+	h.StartWorkers(h.Config.DomainName, ApplicationName, workerOptions)
+}
+
+func startWorkflow(h *common.SampleHelper) *workflow.Execution {
+	workflowOptions := client.StartWorkflowOptions{
+		ID:                              "pageflow_" + uuid.New(),
+		TaskList:                        ApplicationName,
+		ExecutionStartToCloseTimeout:    time.Minute,
```
Comment (mkolodezny) [inline cmd/samples/pageflow/main.go:29]: which state it goes after the timeout?

File: cmd/samples/pageflow/workflow.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+const (
+	Initialized State = "initialized"
+	Received = "received"
+	Created = "created"
+	SubmissionReceived = "submission-received"
+	Submitted = "submitted"
+	Approved = "approved"
+	Rejected = "rejected"
+	Withdrawn = "withdrawn"
```
Comment (mkolodezny) [inline cmd/samples/pageflow/workflow.go:34]: is it the timeout one?

Comment (meiliang86) [reply cmd/samples/pageflow/workflow.go:34]: Timeout will leave workflow at the latest status. So in practice, this need to work with workflow close status, i.e. completed/timed_out/etc.

Comment (mkolodezny) [reply cmd/samples/pageflow/workflow.go:34]: I see, makes sense.

**Pull request merged**